metrics/metrics.go

```diff
@@ -72,33 +73,33 @@ func ensureNotFrozen() {
 	}
 }
 
-// RegisterRootScope initializes the root scope for all the service metrics
-func RegisterRootScope(rep ScopeFunc) {
+// RegisterStatsReporter initializes the stats reporter for all the service metrics
+func RegisterStatsReporter(repFunc StatsReporterFunc) {
 	ensureNotFrozen()
 
 	_mu.Lock()
 	defer _mu.Unlock()
 
-	if _scopeFunc != nil {
+	if _repFunc != nil {
 		// TODO(glib): consider a "force" flag, or some way to clear out and replace the reporter
 		panic("There can be only one metrics reporter")
 	}
 
-	_scopeFunc = rep
+	_repFunc = repFunc
 }
 
 // RootScope returns the provided stats reporter, or nil
-func RootScope(i ScopeInit) (tally.Scope, io.Closer) {
+func RootScope(i ScopeInit) (tally.Scope, tally.StatsReporter, io.Closer) {
 	_mu.Lock()
 	defer _mu.Unlock()
 
-	if _scopeFunc != nil {
-		rep, closer, err := _scopeFunc(i)
+	if _repFunc != nil {
+		rep, err := _repFunc(i)
 		if err != nil {
 			panic(fmt.Sprintf("Failed to initialize metrics reporter %v", err))
 		}
-		return rep, closer
+		scope, closer := tally.NewRootScope("", nil, rep, time.Second)
```

Comment:
This is not what we want, as I was trying to explain earlier. We want to give control over creating this scope to the service owners, if they choose to do so.

The parameters to this function is something that has to be configurable: prefix, tags, and the tick duration.

This is how this file initially looked, by taking in a reporter, but it's a lot less flexible than allowing service owners to create their own metrics structure that might work better for them. We can't make those sort of decisions (especially not the tags they want to include on all metrics events) for them.